$NetBSD: patch-configure,v 1.1 2024/06/18 09:07:25 markd Exp $

c99 removes support for implicit ints, so declare the return type
of main explicitly. From Fedora

fixes for pkgsrc python - from renamed patch-aa

--- configure.orig	2014-09-05 15:08:16.000000000 +0000
+++ configure
@@ -17571,7 +17571,7 @@ $as_echo_n "checking for png_structp in
                   png_structp pp;
 		  png_infop info;
 		  png_colorp cmap;
-		  main() { void*foo = png_create_read_struct; }
+		  int main() { void*foo = png_create_read_struct; }
 _ACEOF
 if ac_fn_c_try_link "$LINENO"; then :
   png_ok=yes
@@ -19308,7 +19308,7 @@ $as_echo_n "checking for $am_display_PYT
 if ${am_cv_python_pythondir+:} false; then :
   $as_echo_n "(cached) " >&6
 else
-  am_cv_python_pythondir=`$PYTHON -c "from distutils import sysconfig; print sysconfig.get_python_lib(0,0,prefix='$PYTHON_PREFIX')" 2>/dev/null ||
+  am_cv_python_pythondir=`$PYTHON -c "from distutils import sysconfig; print sysconfig.get_python_lib(0,0)" 2>/dev/null ||
      echo "$PYTHON_PREFIX/${py_lib}/python$PYTHON_VERSION/site-packages"`
 fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $am_cv_python_pythondir" >&5
@@ -19326,7 +19326,7 @@ $as_echo_n "checking for $am_display_PYT
 if ${am_cv_python_pyexecdir+:} false; then :
   $as_echo_n "(cached) " >&6
 else
-  am_cv_python_pyexecdir=`$PYTHON -c "from distutils import sysconfig; print sysconfig.get_python_lib(1,0,prefix='$PYTHON_EXEC_PREFIX')" 2>/dev/null ||
+  am_cv_python_pyexecdir=`$PYTHON -c "from distutils import sysconfig; print sysconfig.get_python_lib(1,0)" 2>/dev/null ||
      echo "${PYTHON_EXEC_PREFIX}/${py_lib}/python${PYTHON_VERSION}/site-packages"`
 fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $am_cv_python_pyexecdir" >&5
@@ -19420,18 +19420,18 @@ else
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking for libpython${PYTHON_VERSION}.a" >&5
 $as_echo_n "checking for libpython${PYTHON_VERSION}.a... " >&6; }
 
-  py_config_dir="$py_prefix/$py_lib/python${PYTHON_VERSION}/config"
+  py_config_dir="$py_prefix/lib/python${PYTHON_VERSION}/config"
 
   py_makefile="${py_config_dir}/Makefile"
   if test -f "$py_makefile"; then
       py_basemodlibs=`sed -n -e 's/^BASEMODLIBS=\(.*\)/\1/p' $py_makefile`
     py_other_libs=`sed -n -e 's/^LIBS=\(.*\)/\1/p' $py_makefile`
 
-      if test -e "$PYTHON_PREFIX/${py_lib}/libpython${PYTHON_VERSION}.so"; then
+      if test -e "$PYTHON_PREFIX/lib/libpython${PYTHON_VERSION}.so"; then
       PYTHON_LIBS="-L${py_config_dir} -lpython${PYTHON_VERSION} $py_basemodlibs $py_other_libs"
       { $as_echo "$as_me:${as_lineno-$LINENO}: result: found" >&5
 $as_echo "found" >&6; }
-    elif test -e "${py_prefix}/${py_lib}/libpython${PYTHON_VERSION}.a"; then
+    elif test -e "${py_prefix}/lib/libpython${PYTHON_VERSION}.a"; then
             PYTHON_LIBS="-L${py_config_dir} -lpython${PYTHON_VERSION} $py_basemodlibs $py_other_libs"
       { $as_echo "$as_me:${as_lineno-$LINENO}: result: found" >&5
 $as_echo "found" >&6; }
